python/ray/serve/utils.py

```diff
@@ -301,9 +301,26 @@ def try_schedule_resources_on_nodes(
         for node_id, node_resource in ray_resource.items():
             # Check if we can schedule on this node
             feasible = True
-            for key, count in resource_dict.items():
-                if node_resource.get(key, 0) - count < 0:
-                    feasible = False
+
+            # Legacy memory is treated in 50 byte chunks
+            if int(ray.__version__.split(".")[0]) >= 1:
```

Comment:
no need to check the ray version here because I think newer serve version that perform these checks won't run on pre 1.0 ray due to API changes.